data/lib/remote/fetch.py

```
@@ -0,0 +1,151 @@
+        raise FileNotFoundError(
+            f"Remote state files not found in {remote_state_dir}: {', '.join(missing)}"
+        )
+
+    index: dict[str, object] = json.loads(paths["index"].read_text(encoding="utf-8"))
+    docs: dict[str, object] = json.loads(paths["documents_catalog"].read_text(encoding="utf-8"))
+    bundles: dict[str, object] = json.loads(paths["bundles_catalog"].read_text(encoding="utf-8"))
+    return index, docs, bundles
+
+
+def _run(cmd: list[str], redacted_cmd: list[str], title: str) -> None:
+    out = subprocess.run(cmd, capture_output=True, text=True, encoding="utf-8", errors="replace")
```

Ruff (0.15.14)

[error] `subprocess` call: check for execution of untrusted input

(S603)